### ISSUE TITLE
do schemas need to be updated

### ISSUE THREAD

**Issue by nsheff** [databio/bbconf#20]
There are several schemas here: 

https://github.com/databio/bbconf/tree/dev/bbconf/schemas

I believe these are pipestat schemas. But when pipestat was updated, the schema definition changed.

bbconf has I believe been updated to use the new pipestat.

Are these schemas up-to-date?

**Comment by khoroshevskyi** [comment]
This pipestat sachems were updated by @donaldcampbelljr , and I believe they are working and up-to-date

**Comment by donaldcampbelljr** [comment]
I believe I only updated the pipeline names but did not touch the actual schema fields as they were working during testing with the new pipestat. Pipestat should raise a schema error if the required keys are not present for complex types.

**Comment by nsheff** [comment]
I thought the pipestat schema format changed like this:

https://github.com/pepkit/pipestat/issues/20

**Comment by nsheff** [comment]
@donaldcampbelljr pipestat schemas are now json schemas, right?

**Comment by donaldcampbelljr** [comment]
Pipestat can accept either the old way or a JSON schema, yes.

However, after our discussion ~ last week, I realized we are still using types such as file and image within the JSON pipestat schema (and later converting them to be objects). 

Therefore the schema is still not actually 100% JSON schema: https://json-schema.org/understanding-json-schema/reference/type

**Comment by nsheff** [comment]
I thought we decided on doing:

```
type: object
object_type: file
```

that makes it a valid json-schema, but allows our type-specific treatment.

**Comment by donaldcampbelljr** [comment]
Close, but right now its:
```
type: file
object_type: file
```
Pipestat still uses the `type` field and if its an image or a file, recursively replaces that with `title` and `path` and changes the type to object.

**Comment by donaldcampbelljr** [comment]
Current bbconf example from dev:

```
pipeline_name: bedfile
samples:
  name:
    type: string
    description: BED file name
  genome:
    type: object
    description: genome assembly of the BED files
  bedfile:
    type: file
    pipestat_type: file
    label: bed
    description: BED file
  bigbedfile:
    type: file
    pipestat_type: file
    label: bigBed
    description: bigBed file
```

**Comment by nsheff** [comment]
that's the old format, not a json schema at all

**Comment by donaldcampbelljr** [comment]
Yes, pipestat can accept either format currently. However, I'm just showing how the pipestat_type was implemented along with the original type.

**Comment by nsheff** [comment]
We need to update those bbconf schemas to use the new format, and deprecate the old.

Issue raised: https://github.com/databio/bbconf/issues/32

**Comment by nsheff** [comment]
The relevant issue in pipestat is: https://github.com/pepkit/pipestat/issues/85

**Comment by khoroshevskyi** [comment]
Fixed in 0.4.0 release
